app/controllers/users_controller.rb

```diff
@@ -1,5 +1,5 @@
 class UsersController < ApplicationController
-  # GET /users or /users.json
+  before_action :authenticate_user!
```

Comment:
**OPTIONAL** - Though I know the previous reviewer asked you to include this here, but it'd be better if you added this in your application controller instead. Currently only the actions in your user controller will be authenticated, I can access all actions in the posts controller without authentication. Please fix this 👍 👍